lazy_dataset/core.py

```diff
@@ -338,6 +338,18 @@ def indexable(self) -> bool:
 
     @property
     def ordered(self) -> bool:
+        """
+        Indicate whether each iteration over the dataset yield the examples
+        in the same order.
+
+        This property is used for the `cache` implementation.
+
+        Note: This property indicates only the ordered property from the
+              dataset side.
+              When a map function is not deterministic, i.e. has some random
+              parts, this property should be false, but the dataset is unable
+              to know this.
```

Comment:
In my opinion, `ordered` doesn't/shouldn't tell anything about the contents of the dataset or whether it is deterministic. It just represents that you get the examples in the same order when you iterate multiple times over the dataset. But, it is true that the cache dataset can't detect this. I wouldn't add this comment here but in the `cache` method/dataset.

Reply:
I think a repeat of this information here is important.
When you implement new datasets, it is important to know what this property represents and for what it is used.